src/user.rs

```diff
@@ -0,0 +1,177 @@
+//! Tasks defined rust_actix-basic.pdf
+//! Using endpoints at: https://jsonplaceholder.typicode.com/<API>
+
+use validator::Validate;
+use validator_derive;
+
+use reqwest;
+use serde::{Deserialize, Serialize};
+
+/// Submits a GET request to a specified URL, and returns a vector of users. Expected response is in json format, and can be parsed by the `User` struct with accessible elements by key name.
+pub async fn get_users(url: &str) -> Result<Vec<User>, Box<dyn std::error::Error>> {
+    let users = reqwest::get(url).await?.json::<Vec<User>>().await?;
+
+    Ok(users)
+}
+
+/// `POST` a new user with a JSON formatted raw string (&str) passed into `post_user`. Returns a `User` struct with fields filled as returned by the endpoint,if the post is valid. This will include a new `User.id` field if the user is set properly.
+///
+/// The name will need to be at least 3 characters long and the email address should be valid.
+///
+/// ## Errors
+///
+/// This will panic if the `name` is less than 3 characters, or `email` is not valid.
+///
+/// > Note: The name will need to be at least 3 characters long and the email address should be valid.
+pub async fn post_user(url: &str, user_json: &str) -> Result<User, Box<dyn std::error::Error>> {
```

Comment:
Try renaming `user_json` to just `user`, and create a struct for the post (`CreateUserRequest`).  The full signature of `user` should be: `Json<CreateUserRequest>`